op-monitorism/transaction_monitor/checks.go

```diff
@@ -0,0 +1,120 @@
+package transaction_monitor
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+// CheckFunc is a type for address verification functions
+type CheckFunc func(ctx context.Context, client *ethclient.Client, addr common.Address, params map[string]interface{}) (bool, error)
+
+// AddressChecks maps check types to their implementation functions
+var AddressChecks = map[CheckType]CheckFunc{
+	ExactMatchCheck:  CheckExactMatch,
+	DisputeGameCheck: CheckDisputeGame,
+}
+
+// CheckExactMatch verifies if the address matches exactly with the provided match parameter
+func CheckExactMatch(ctx context.Context, client *ethclient.Client, addr common.Address, params map[string]interface{}) (bool, error) {
+	match, ok := params["match"].(string)
+	if !ok {
+		return false, fmt.Errorf("match parameter not found or invalid")
+	}
+	return addr == common.HexToAddress(match), nil
+}
+
+// CheckDisputeGame verifies if the address is a valid dispute game created by the factory
+func CheckDisputeGame(ctx context.Context, client *ethclient.Client, addr common.Address, params map[string]interface{}) (bool, error) {
+	// First check if there's any code at the address
+	code, err := client.CodeAt(ctx, addr, nil)
+	if err != nil {
+		return false, fmt.Errorf("failed to get code at address: %w", err)
+	}
+	if len(code) == 0 {
+		return false, nil
+	}
+
+	// Get factory address from params
+	factoryAddr, ok := params["disputeGameFactory"].(string)
```

Comment:
On second thought it's probably best if we have a function like `VerifyCheckExactMatchParams` and `VerifyCheckDisputeGameParams` so that we can run these functions at initialization time and not have these errors trigger later at runtime